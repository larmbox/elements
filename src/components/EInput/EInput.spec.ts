import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { EInput } from '~/index';

test('mount component', () => {
  expect(EInput).toBeTruthy();

  const component = mount(EInput, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('EInput');
  expect(component.html()).toContain('EFormControl--field');
});

test('register css variables', () => {
  mount(EInput);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('EInput');
});

test('inherit attributes', () => {
  const component = mount(EInput, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
