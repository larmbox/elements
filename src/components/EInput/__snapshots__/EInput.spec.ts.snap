// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl EFormControl--field EInput EInput--md EInput--primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"EInput-inner\\">
    <div class=\\"EInput-element\\"><input id=\\"id\\" type=\\"text\\" /></div>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
